packages/wallet/src/KeyManagement/types.ts

```diff
@@ -138,8 +138,15 @@ export interface Ed25519KeyPair {
  */
 export type GetPassword = (noCache?: true) => Promise<Uint8Array>;
 
+/**
+ * @returns A Ed25519 transaction signature.
+ */
+// eslint-disable-next-line no-use-before-define
+export type SignTx = (txInternals: TxInternals, options?: SignTransactionOptions) => Promise<Cardano.Signatures>;
```

Comment:
If the goal is for this to be compatible with KeyAgent's `signTransaction`, then this might be slightly better:
```suggestion
export type SignTx = KeyAgent['signTransaction'];
```

Reply:
There's a problem with the refactor: it will now call `signTransaction` for every extra key path, which will result in bad UX (e.g. for InMemoryKeyAgent it will call `getPassword` multiple times). Original implementation with `additionalKeyPaths` only needed to decrypt root private key once to sign with all key paths.

Maybe let's consider `extraSigners` only for external keys, and if the goal is to use the same wallet's key agent then user is supposed to use 'additionalKeyPaths'?

Reply:
The extra signatures will probably come from different key agents, but I do understand what you mean, if the same key agent is performing all the signatures it will have to authorize several times.

I will add the SignTransactionOptions so the user can choose how to add the extra signatures, via adding additional key paths for the signing operation or by adding extra signers

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/404/commits/4ec4aadd1582e7a836d93599be9e2ddad4f556c3